chainer/functions/activation/softmax.py

```diff
@@ -74,22 +74,38 @@ def backward(self, x, gy):
 
 
 def softmax(x, use_cudnn=True):
-    """Channelwise softmax function.
+    """Channel-wise softmax function.
 
     This function computes its softmax along the second axis. Let
-    :math:`x = (x_1, x_2, \\dots, x_d)^{\\top}` be the d dimensional index
-    array and :math:`f(x)` be the d dimensional input array. For each index
-    :math:`x` of the input array :math:`f(x)`, it computes the probability
-    :math:`p(x)` defined as
-    :math:`p(x) = {\\exp(f(x)) \\over \\sum_{x_2} \\exp(f(x))}`.
+    :math:`x=(x_1, x_2, \\dots, x_D)^{\\top}` be the D dimensional
+    input array.
+    For each input array :math:`x`, it computes the function :math:`f(x)`
+    defined as
+    :math:`f(x)={\\exp(x) \\over \\sum_{x_d} \\exp(x_d)}`.
 
     Args:
-        x (~chainer.Variable): Input variable.
+        X (:class:`~chainer.Variable` or :class:`numpy.ndarray` or \
+        :class:`cupy.ndarray`):
+            Input variable.
+            A 2d (N, D) :math:`((s_1^1, ..., s_D^1), ..., (s_1^N, ..., s_D^N))`
+            -shaped float array.
         use_cudnn (bool): If ``True`` and cuDNN is enabled, then this function
             uses cuDNN as the core implementation.
 
     Returns:
-        ~chainer.Variable: Output variable.
+        ~chainer.Variable: Output variable. A 2d (N, D)
+        :math:`((s_1^1, ..., s_D^1), ..., (s_1^N, ..., s_D^N))`
+        -shaped float array.
+
+    .. admonition:: Example
+
+        >>> x = np.array([[0, 1, 2], [0, 2, 4]], 'f')
+        >>> x
+        array([[ 0.,  1.,  2.],
+               [ 0.,  2.,  4.]], dtype=float32)
+        >>> F.softmax(x).data
+        array([[ 0.09003057,  0.24472848,  0.66524094],
+               [ 0.01587624,  0.11731043,  0.86681336]], dtype=float32)
```

Comment:
How about showing that the summation over the second axis is always 1? For example,
```
>>> y = F.softmax(x)
>>> y.data
array([[ 0.09003057,  0.24472848,  0.66524094],
       [ 0.01587624,  0.11731043,  0.86681336]], dtype=float32)
>>> F.sum(y, axis=1).data
array([ 1.,  1.], dtype=float32)
```